### PR TITLE
feat: Clean up shipping options management

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
- cleanup existing route on some aspects but not all
- add new create shipping options end point and types
- add set of integration tests
- cleanup existing fulfillment routes
- align existing integration tests
- transform old type to zod types and validators
- fix stock location route

### CODE REVIEW

File: packages/medusa/src/api-v2/admin/fulfillment/middlewares.ts [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+
+  {
+    method: ["POST"],
+    matcher: "/admin/fulfillment/shipping-options",
+    middlewares: [
+      transformQuery(
+        AdminPostShippingOptionsShippingOptionParams,
+        retrieveTransformQueryConfig
+      ),
+      transformBody(AdminPostFulfillmentShippingOptionsShippingOption),
```
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/fulfillment/middlewares.ts]: suggestion: Use zod

File: packages/medusa/src/api-v2/admin/fulfillment/validators.ts [hunk 2]
```diff
@@ -34,3 +51,86 @@ export class FulfillmentRuleCreate {
...
+export class AdminPostFulfillmentShippingOptionsShippingOptionRegionPrice {
+  @IsString()
+  @IsNotEmpty()
+  region_id: string
+
+  @IsNotEmpty()
+  amount: number
+}
+
+export class AdminPostFulfillmentShippingOptionsShippingOption {
```
Comment (olivermrbl) [inline packages/medusa/src/api-v2/admin/fulfillment/validators.ts]: suggestion: use zod

File: packages/medusa/src/api-v2/admin/fulfillment/middlewares.ts (hunk 1 above)
Comment (adrien2p) [reply packages/medusa/src/api-v2/admin/fulfillment/middlewares.ts]: true, I ll change that

File: integration-tests/modules/__tests__/fulfillment/admin/shipping-options.spec.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        })
+
+        region = await regionService.create({
+          name: "Test region",
+          countries: ["FR"],
+          currency_code: "eur",
+        })
+      })
+
+      describe("POST /admin/fulfillment/shipping-options", () => {
```
Comment (olivermrbl) [inline integration-tests/modules/__tests__/fulfillment/admin/shipping-options.spec.ts]: The endpoint should be `/admin/shipping-options`

Comment (adrien2p) [reply integration-tests/modules/__tests__/fulfillment/admin/shipping-options.spec.ts]: should I change the other existing end points as well? I ve tried to keep it consistent

File: packages/types/src/http/pricing/price.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+export interface AdminPrincingPriceSetPriceResponse {
```
Comment (olivermrbl) [inline packages/types/src/http/pricing/price.ts]: nit:
```suggestion
export interface AdminPriceSetPriceResponse {
```

**Review by olivermrbl — APPROVED** [review]
💪 💪 💪

**Review by sradevski — APPROVED** [review]
Looking good!

**Pull request merged**